### ISSUE TITLE
Module build failed (from ./node_modules/fable-loader/index.js): Error: File src/App.fs cannot be found in source list

### ISSUE THREAD

**Issue by entyo** [fsprojects/FSharpPlus#243]
### Description

Checking Release Notes, I noticed that Fable support of FSharpPlus has been enhanced in recent versions. So I tried to use `1.1.0-RC1` in fable-loader project.

I just wrote some code in `src/App.fs`:

```fsharp
module App

open Fable.Core.JS
open FSharpPlus

let res1 = map ((+) 2) (Some 2)

console.log "hello"
```

But when I try to transpile, I got the error in the title.

```
ERROR in ./src/App.fs
Module build failed (from ./node_modules/fable-loader/index.js):
Error: File /home/e_ntyo/fable-fsharpplus-sample/src/App.fs cannot be found in source list:
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Internals.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Extensions.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Converter.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Numeric.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Monoid.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Functor.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Foldable.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Bifoldable.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Traversable.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Indexable.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/MonadTrans.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Collection.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Tuple.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Operators.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Builders.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Identity.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/List.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Seq.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Option.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Error.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Cont.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Reader.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Writer.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/State.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Monoids.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Lens.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/NonEmptyList.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/ZipList.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/ParallelArray.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/DList.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Validations.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Kleisli.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Free.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Coproduct.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Memoization.fs
/home/e_ntyo/fable-fsharpplus-sample/.fable/FSharpPlus.1.1.0-RC1/Parsing.fs (/home/e_ntyo/fable-fsharpplus-sample/src/App.fs)
    at Loader.command.then.data (/home/e_ntyo/fable-fsharpplus-sample/node_modules/fable-loader/index.js:98:22)
    at process._tickCallback (internal/process/next_tick.js:68:7)
 @ ./src/App.fsproj 1:0-25 1:0-25
```

The exception seems to be thrown in [`transformFile` function in `FSharp2Fable.fs`](https://github.com/fable-compiler/Fable/blob/5d9c9ed1d7f56f195bfd9696420e22f1bd61a237/src/Fable.Transforms/FSharp2Fable.fs#L1166), which is a part of `fable-loader`.

### Repro steps

I created a sample project for repro at [here](https://github.com/entyo/fable-fsharpplus-sample).

3 STEPS TO RUN:

1. `yarn install`
2. `paket install`
3. `yarn start`

### Expected behavior

`App.fs` should be transpiled successfully.

### Actual behavior

I got `Error: File /home/e_ntyo/fable-fsharpplus-sample/src/App.fs cannot be found in source list:` and transpilation failed.

### Related information 

My runtime environment:

- .NET Core 3.1
- Paket 5.241.6

**Comment by gusty** [comment]
Thanks for the repro. I'll try to find out what's wrong.

**Comment by wallymathieu** [comment]
Seems like you can use `Map.Invoke` as a workaround for the issue.

**Comment by entyo** [comment]
@wallymathieu Thank you ! I checked this commit https://github.com/fsprojects/FSharpPlus/pull/244/commits/41fb254e5f1b88b6fb6136693d4285767f530f30

**Comment by wallymathieu** [comment]
This sounds related: https://github.com/fable-compiler/Fable/issues/1940

**Comment by gusty** [comment]
OK, now I think we can explain a bit this issue.

 - Fable compilation was broken and the test we had at the moment of this issue were not checking the full compilation.

 - I think there might be a bug in the Fable side that when certain errors are present on the source files it breaks something internally in the compiler and it reports as the file (with the error) is missing. ( cc @alfonsogarciacaro  ). I encountered that situation all the time while fixing the fable conditionals.

 - At the moment of writing this, Fable doesn't seems to support complex trait calls likes the one used by the `map` function, so for the moment `map` is not supported in Fable. We might add partial support soon so it will become usable to some extent.

 - I suggest you try the recently released v1.1.0-RC3 which has proper testing, although `map` is not supported, but `>>=` is (see the release notes). Feedback is more than welcome and feel free to re-open if the problem persist.
